### PR TITLE
add zope and setuptools deps, pip check

### PR DESCRIPTION
**Pull request by bollwyvl**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

notes:
- adds `zope.interface`, `zope.events` and `setuptools` to run dependencies
  - as introduced in https://github.com/gevent/gevent/pull/1620
- adds `pip check` to test to avoid future surprises
- unblocks https://github.com/conda-forge/dagster-feedstock/pull/36

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -35,21 +35,29 @@ requirements:
...
     - cffi >=1.11.5
     - python
     - greenlet >=0.4.14
+    - zope.event
+    - zope.interface
+    - setuptools
 
 test:
+  requires:
+    - pip
```
Comment (bollwyvl) [inline recipe/meta.yaml:51]: yeah, going to need `pip` to `pip` check

**Comment by bollwyvl** [comment]
I took a crack at running the tests: a fair amount of stuff broke, and even revealed some _more_ dependency issues, but there were a large number of fails, so probably not worth delaying getting the current (and future) declared dependencies satisfied...

**Comment by bollwyvl** [comment]
(test dependency issue in question: https://github.com/conda-forge/objgraph-feedstock/pull/1)

**Comment by bollwyvl** [comment]
Looks like locally i pulled `zope.event` from `anaconda`... probably need that!

**Comment by bollwyvl** [comment]
yup:
```
    zope.event:      4.4-py37_0                defaults   
    zope.interface:  5.1.0-py37h9bfed18_0      conda-forge

```

**Comment by bollwyvl** [comment]
PR created: https://github.com/conda-forge/staged-recipes/pull/11823

**Comment by bollwyvl** [comment]
Looks like drone isn't restarting automatically right now:
https://cloud.drone.io/conda-forge/gevent-feedstock/28

Apparently just needs a manual kick, or an empty commit...

**Comment by bollwyvl** [comment]
Looks ready for review...

**Comment by nehaljwani** [comment]
Thanks @bollwyvl !

**Pull request merged**